restconf_update_ipaddress/updateip.py

```diff
@@ -142,12 +150,23 @@ def interface_selection(interfaces, mgmt_if):
     return(sel)
 
 
-# Asks the user to provide an IP address and Mask.  Data is NOT validated.
-def get_ip_info():
+# Asks the user to provide an IP address and Mask.
+def get_ip_info(cidr):
     # Ask User for IP and Mask
     ip = {}
-    ip["address"] = input("What IP address do you want to set? ")
-    ip["mask"] = input("What Subnet Mask do you want to set? ")
+    try:
+        if cidr:
+            (ipa_t, ipl) = input("What IP address/prefixlen do you want to set? ").split('/')
+            ipa = ipaddress.ip_address(ipa_t)
+            ip["address"] = ipa.compressed
+            ip["mask"] = ipa._make_netmask(int(ipl))[0].compressed
```

Comment:
I don't like using protected methods like this.  What about `ip_net = ipaddress.ip_network(f"{ipa_t}/{ipl}", strict=False)`.  Then you can get the address like you do and the mask using .netmask.

Reply:
I have some concerns about using ip_network() here.  In this particular example, some information is lost:

>>> ipa_t
'10.2.0.107'
>>> ipl
'24'
>>> ip_net = ipaddress.ip_network(f"{ipa_t}/{ipl}", strict=False)
>>> ip_net
IPv4Network('10.2.0.0/24')

Since this is configuring an interface, as opposed to something like BGP NLRI, I'd like to preserve address semantics, if possible.

Reply:
I didn't mean to change your approach to the address.  You can still do that. Just use the more _public_ approach to net the netmask.

Reply:
I'm curious ... is your problem with _make_netmask() that it is protected, or because you feel it won't work in some way?  If it was public, would it meet your criteria?

There may be a way to do this with ip_interface().  I'll look into it and get back to you.

>>> ip = ipaddress.ip_interface('10.2.0.107/24')
>>> ip
IPv4Interface('10.2.0.107/24')
>>> ip.ip
IPv4Address('10.2.0.107')
>>> ip.ip.compressed
'10.2.0.107'
>>> ip.netmask
IPv4Address('255.255.255.0')
>>> ip.netmask.compressed
'255.255.255.0'

Reply:
That is my concern.  Not that it won't work now, but that it may break in the nearer-term future.

Reply:
I think ip_interface() will do what we want.  Take a look at the latest commit and let me know.